x-pack/legacy/plugins/lens/public/persistence/saved_object_store.ts

```diff
@@ -62,18 +61,17 @@ export class SavedObjectIndexStore implements SavedObjectStore {
 
   async save(vis: Document) {
     const { id, type, ...rest } = vis;
-    const attributes = {
-      ...rest,
-      state: JSON.stringify(rest.state),
-    };
+
+    // Any is the most straighforward way out here, since the saved
+    // object client doesn't allow unknowns, and we have them in
+    // our object.
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const attributes: any = rest;
```

Comment:
Agree, the types of the saved object client can be improved in this regard. However we don't need to fall back to disabling linting rules, the "any-less" way of doing that would be
```
// TODO SavedObjectAttributes should support this kind of object,
// remove this workaround till the type is updated.
const attributes = rest as unknown as SavedObjectAttributes;
```